include/blaze/Bus.hpp

```diff
@@ -15,7 +15,9 @@ namespace Blaze
 		~Bus();
 
 		//=== Bus Fucntionality ===
-		void write(Address addr, Byte data);
-		Byte* read(Address addr);
+		void write(Address addr, Byte data); 	// write8
+		void write(Address addr, Word data); 	// write16
+		void write(Address addr, Address data); // write 24
+		auto read(Address addr);				// Return either Byte*, Word*, or Address*
```

Comment:
You can't do this; the function can't morph return types as necessary; it has to have a specific return type. You can use auto in front, but then you have to follow it with an arrow and the right return type after the parameters, e.g.:

```cpp
auto someFunction(int foo, int bar) -> int;
```

To do what you want to do, you *have* to have 3 separate functions with different names, since you also can't do overloads based on return types.